src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -560,12 +560,88 @@ class IOpenGL_PhysicalDeviceBase : public IOpenGLPhysicalDeviceBase
 		}
 		else m_glfeatures.MaxAnisotropy = 0u;
+		
+		GLint64 maxElementIndex = 0;
+		GetInteger64v(GL_MAX_ELEMENT_INDEX, &maxElementIndex);
+		 
+		m_features.fullDrawIndexUint32 = (maxElementIndex == 0xffff'ffff);
+		m_features.independentBlend = (!IsGLES); // TODO
+
+		if (!IsGLES || m_glfeatures.Version >= 320u)
+		{
+			#define GLENUM_WITH_SUFFIX(X) X
+			#include "nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_OES_tessellation_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_OES
+			#include "nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_EXT_tessellation_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_EXT
+			#include "nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		// else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_INTEL_tessellation_shader))
+		{
+			//!! no _INTEL suffix
+			#define GLENUM_WITH_SUFFIX(X) X##_INTEL
+			// #include "src/nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		
+		if (!IsGLES || m_glfeatures.Version >= 320u)
+		{
+			#define GLENUM_WITH_SUFFIX(X) X
+			#include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_OES_geometry_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_OES
+			#include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_EXT_geometry_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_EXT
+			#include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		// else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_INTEL_geometry_shader))
+		{
+			//!! no _INTEL suffix
+			#define GLENUM_WITH_SUFFIX(X) X##_INTEL
+			// #include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+
+		m_features.drawIndirectFirstInstance = false; // TODO
+		m_features.depthClamp = m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_ARB_depth_clamp);
+		m_features.depthBiasClamp = false; // TODO
 
-		if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_ARB_geometry_shader4))
+		m_features.fillModeNonSolid = false; //TODO
+		m_features.depthBounds = m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_EXT_depth_bounds_test);
+		m_features.wideLines = false; //TODO
+		m_features.largePoints = false; //TODO
+		m_features.alphaToOne = false; //TODO
+		
+		m_features.shaderClipDistance = true;
+		GetIntegerv(GL_MAX_CLIP_DISTANCES, reinterpret_cast<GLint*>(&m_properties.limits.maxClipDistances));
+		if(m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_ARB_cull_distance))
+		{
+			m_features.shaderCullDistance = true;
+			// no ARB suffix, is that ok?
```

Comment:
yeah its fine, none of the headers have an `ARB` define anyway

Reply:
actually this is why
https://github.com/Devsh-Graphics-Programming/Nabla/pull/357/files/43b71f3d6a87d3c944cbcb2375cd9ec157657eb0..18ad613cfde62615bac125ab7adb06cd07e046dc#r893487468